### PR TITLE
[State Management] reconsider state management

### PR DESCRIPTION
**Pull request by lotrien**
- removed redundant Redux store with huge amount of boilerplate in favor of local state with Recoil;
- optimized pagination request handling with Recoil (it caches prev results and doesn't make additional request for the same markers);
- optimized `/:snippetId` route by checking cached snippet by id and if it's not there -> make request

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,7 +85,7 @@
     "webpack-dev-server": "^3.11.1"
   },
   "scripts": {
-    "lint": "eslint --ignore-path .gitignore --ext js,jsx .",
+    "lint": "eslint --ignore-path .gitignore --ext js,jsx,tsx .",
```
Comment (Snizhana) [inline package.json]: 👍 we should probably add `.ts` as well

File: src/api/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { getApiUri } from '../misc/url'
+import { Snippet, RawSnippet } from '../store'
+
+export const fetchSnippet = (id: number) => {
```
Comment (Snizhana) [inline src/api/index.ts]: nit: we can add a return type to these helpers

File: src/api/index.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  return fetch(getApiUri('snippets'), {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(snippet),
+  })
+    .then(response => response.json())
+    .then((json) => {
```
Comment (Snizhana) [inline src/api/index.ts]: minor, but there is no need to create an anonymous function `.then(onSuccess)` should work

File: src/misc/modes.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export const normalizedSyntaxes = (syntaxes: string[]) => {
+  const { modesByName } = getModesByName()
+
+  return syntaxes.map(item => ({
+    name: modesByName[item].caption,
+    value: item,
+  }))
+}
```
Comment (Snizhana) [inline src/misc/modes.ts]: minor, a blank line is missing

**Review by Snizhana — DISMISSED** [review]
wow, it is so nice to see 800 lines of removed code 😊

**Comment by lotrien** [comment]
hey guys @Snizhana @ikalnytskyi I would highly appreciate some 👀 on recoil usage

File: package.json (hunk 1 above)
Comment (lotrien) [reply package.json]: I tend to add this in separate PR, it seems there is a need to add some additional packages in order for lint to work properly with `.ts` files. Don't want to make a mess here

**Review by ikalnytskyi — APPROVED** [review]
I have no idea what Recoil is, but looks better now! :muscle:

**Pull request merged**